### PR TITLE
[MISC] pass args instead of individual arguments

### PR DESCRIPTION
**Pull request by Irallia**
Resolves first part of https://github.com/seqan/iGenVar/issues/87 

This PR is mostly moving functions from one file into another.

### CODE REVIEW

**Review by MitraDarja — APPROVED** [review]

File: include/variant_detection/method_enums.hpp [hunk 1]
```diff
@@ -32,3 +35,21 @@ enum refinement_methods
     sViper_refinement_method = 1,
     sVirl_refinement_method = 2
 };
+
+/*! \brief Specialise a mapping from an identifying string to the respective value of your type methods. With the help
```
Comment (eldariont) [inline include/variant_detection/method_enums.hpp:39]:
```suggestion
/*! \brief Specialise a mapping from an identifying string to the respective value of your type detection_methods. With the help
```

File: src/iGenVar.cpp [hunk 2]
```diff
@@ -1,72 +1,9 @@
...
-{
-    std::filesystem::path alignment_short_reads_file_path{""};
-    std::filesystem::path alignment_long_reads_file_path{""};
-    std::filesystem::path insertion_file_path{};
-    std::filesystem::path output_file_path{};
-    std::vector<detection_methods> methods{cigar_string, split_read, read_pairs, read_depth};   // default: all methods
-    clustering_methods clustering_method{simple_clustering};                                    // default: simple clustering method
-    refinement_methods refinement_method{no_refinement};                                        // default: no refinement
-    uint64_t min_var_length = 30;
-};
```
Comment (eldariont) [inline src/iGenVar.cpp:68]: Why do you move the `cmd_arguments` struct to `variant_detection.hpp`? Wasn't it quite logical to have it close to the argument parsing method?

**Review by eldariont — APPROVED** [review]
Looks great. Only two smaller things..

File: src/iGenVar.cpp (hunk 2 above)
Comment (Irallia) [reply src/iGenVar.cpp:68]: I will move it back in the next PR (#104) it was necessary as `cmd_arguments` is not known in `variant_detection.hpp`, but in the next PR I'll move the whole function into iGenVar.cpp

Comment (Irallia) [reply src/iGenVar.cpp:68]: I added a TODO for this.

**Pull request merged**